src/bindings/swig/python3/kdb.i

```diff
@@ -38,54 +42,45 @@
 
 // constructors
 %pythonprepend kdb::Key::Key {
-  orig = []
-  if len(args):
-    orig = args[1:]
-    args = [ args[0] ]
+  value = None
+  meta  = {}
+  # check for copy constructor
+  if len(args) and not isinstance(args[0], self.__class__):
+    arg0, args = args[0], args[1:]
+
+    flags = 0
+    args = iter(args)
+    for arg in args:
+      if arg == KEY_END:
+        break
+      elif arg == KEY_SIZE:
+        # ignore value
+        next(args)
+      elif arg == KEY_VALUE:
+        value = next(args)
+      elif arg == KEY_FUNC:
```

Comment:
The bindings share the documentation as well as the enums, thus KEY_FUNC is available. The user will either get a TypeError (the usual python-way) or a "Unknown option"-warning. I prefer the first.